app/ante/fee_checker.go

```diff
@@ -1,54 +1,84 @@
 package ante
 
 import (
-	"fmt"
-
 	errors "cosmossdk.io/errors"
+	"cosmossdk.io/math"
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
 	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
+	"github.com/celestiaorg/celestia-app/x/minfee"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerror "github.com/cosmos/cosmos-sdk/types/errors"
+	params "github.com/cosmos/cosmos-sdk/x/params/keeper"
 )
 
 const (
 	// priorityScalingFactor is a scaling factor to convert the gas price to a priority.
 	priorityScalingFactor = 1_000_000
 )
 
-// CheckTxFeeWithGlobalMinGasPrices implements the default fee logic, where the minimum price per
-// unit of gas is fixed and set globally, and the tx priority is computed from the gas price.
-func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+// CheckTxFeeWithMinGasPrices implements default fee validation logic for transactions.
+// It ensures that the provided transaction fee meets a minimum threshold for the validator
+// as well as a global minimum threshold and computes the tx priority based on the gas price.
+func CheckTxFeeWithMinGasPrices(ctx sdk.Context, tx sdk.Tx, paramKeeper params.Keeper) (sdk.Coins, int64, error) {
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
 		return nil, 0, errors.Wrap(sdkerror.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
 	fee := feeTx.GetFee().AmountOf(appconsts.BondDenom)
 	gas := feeTx.GetGas()
-	appVersion := ctx.BlockHeader().Version.App
 
-	// global minimum fee only applies to app versions greater than one
-	if appVersion > v1.Version {
-		globalMinGasPrice := appconsts.GlobalMinGasPrice(appVersion)
+	// Ensure that the provided fee meets a minimum threshold for the validator.
+	// This is only for local mempool purposes, and thus
+	// is only ran on check tx.
+	if ctx.IsCheckTx() {
+		minGasPrice := ctx.MinGasPrices().AmountOf(appconsts.BondDenom)
+		if !minGasPrice.IsZero() {
+			err := verifyMinFee(fee, gas, minGasPrice, "insufficient minimum gas price for this validator")
```

Comment:
[optoinal]

since all nodes will run this, validator might be confusing to users